### PR TITLE
Item 6583: Update ExpDataFile converter file path resolution

### PR DESCRIPTION
**Pull request by labkey-martyp**
When resolving full path for relative path in dataFileUrl, the file root for Windows does not have a leading "/" and Unix does have a leading "/".  Update to only add this leading slash if it does not exist.

### CODE REVIEW

**Review by labkey-kevink — APPROVED** [review]
Looks ok to me, but I'd have @labkey-jeckels take a look as well.

**Review by labkey-jeckels — APPROVED** [review]
This change seem safe, but I'd really like to push more of these consistency enforcement codepaths inside  getFileRootPath() (and similar methods) so that we don't have to make all of the callers deal with such a variety of paths

File: api/src/org/labkey/api/data/ExpDataFileConverter.java [hunk 1]
```diff
@@ -129,7 +129,15 @@ else if (dataObject.has(ExperimentJSONConverter.DATA_FILE_URL) && dataObject.get
...
-                    data = expSvc.getExpDataByURL(FileUtil.createUri("/" + root + dataFileURL).toString(), container);
+
+                    // non-unix needs to prepend with a slash
+                    if (!root.startsWith("/") && !root.startsWith("\\"))
+                    {
+                        root = "/" + root;
+                    }
+
+                    String path = FileUtil.createUri(root + dataFileURL).toString();
+                    data = expSvc.getExpDataByURL(path, container);
```
Comment (labkey-tchad) [inline api/src/org/labkey/api/data/ExpDataFileConverter.java]: This was the change I mentioned earlier (requires `java.net.URI` to be imported as well)
```suggestion
                    java.nio.file.Path root = FileContentService.get().getFileRootPath(container, FileContentService.ContentType.files);
                    URI dataFileUri = root.resolve(dataFileURL).toUri();
                    data = expSvc.getExpDataByURL(dataFileUri.toString(), container);
```

Comment (labkey-martyp) [reply api/src/org/labkey/api/data/ExpDataFileConverter.java]: This didn't work for me at least on Windows.  Tried several different approaches to get the same result for Linux and Windows but only way that worked was string manipulation.

**Pull request merged**